include/nbl/builtin/hlsl/bda/__ptr.hlsl

```diff
@@ -0,0 +1,37 @@
+// Copyright (C) 2018-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#include "nbl/builtin/hlsl/bda/__ref.hlsl"
+
+#ifndef _NBL_BUILTIN_HLSL_BDA_PTR_INCLUDED_
+#define _NBL_BUILTIN_HLSL_BDA_PTR_INCLUDED_
+
+namespace bda
+{
+template<typename T, bool _restrict>
+struct __ptr
+{
+    using this_t = __ptr < T, _restrict>;
+    uint64_t addr;
+
+    static this_t create(const uint64_t _addr)
+    {
+        this_t retval;
+        retval.addr = _addr;
+        return retval;
+    }
+
+    template<uint32_t alignment>
```

Comment:
you cna give a default value, we actually have an `alignof()` in Nabla's hlsl, include the `type_traits,hlsl` header and see

Reply:
Done

Reply:
Done